lib/iris/fileformats/um/_fast_load_structured_fields.py

```diff
@@ -99,6 +99,10 @@ def core_data(self):
         return self.data
 
     @property
+    def data_dtype(self):
+        return np.result_type(*[field.data_dtype for field in self._fields])
```

Comment:
I think it's ok. `data_dtype` is used in place of `core_data().dtype`. If you look at how that is derived, it's basically by stacking the underlying fields' data arrays together. The result of that stacking has the same dtype as `np.result_type(...)` gives.